recipes-extended/xen/files/defconfig

```diff
@@ -29,19 +42,28 @@ CONFIG_GRANT_TABLE=y
 # CONFIG_GRANT_TABLE_EXOTIC is not set
 CONFIG_HAS_ALTERNATIVE=y
 CONFIG_HAS_EX_TABLE=y
+CONFIG_HAS_FAST_MULTIPLY=y
 # CONFIG_MEM_ACCESS_ALWAYS_ON is not set
 # CONFIG_MEM_ACCESS is not set
-# CONFIG_HAS_MEM_PAGING is not set
-# CONFIG_HAS_MEM_SHARING is not set
+CONFIG_HAS_MEM_PAGING=y
 CONFIG_HAS_PDX=y
-# CONFIG_HAS_UBSAN is not set
-# CONFIG_HAS_KEXEC is not set
-# CONFIG_HAS_GDBSX is not set
+CONFIG_HAS_UBSAN=y
+CONFIG_HAS_KEXEC=y
 CONFIG_HAS_IOPORTS=y
+CONFIG_HAS_SCHED_GRANULARITY=y
 CONFIG_NEEDS_LIBELF=y
+
+#
+# Speculative hardening
+#
+CONFIG_SPECULATIVE_HARDEN_ARRAY=y
+CONFIG_SPECULATIVE_HARDEN_BRANCH=y
+# end of Speculative hardening
+
+CONFIG_HYPFS=y
```

Comment:
If we compile with this on now, do you have another commit that is mounting/using the hypervisor file system?  Is this something we maybe shouldn't provide from a security perspective?

Reply:
That's a good question.  While it has FS in the name, it's not a mountable FS.  But it is a way to expose some stuff from the hypervisor to ~dom0.  XSM restricts it to dom0_type only, so I think it's okay.

    Remove XEN_SYSCTL_set_parameter
    
    xen commit 02e9a9cf2095 "xen: remove XEN_SYSCTL_set_parameter support"
    says "The functionality of XEN_SYSCTL_set_parameter is available via
    hypfs now, so it can be removed."  Mirror the change.

So I think it may be needed in place of the above?